roomserver/internal/query/query.go

```diff
@@ -858,6 +858,52 @@ func (r *Queryer) QueryAuthChain(ctx context.Context, req *api.QueryAuthChainReq
 	return nil
 }
 
+func (r *Queryer) IsInvitePending(ctx context.Context, roomID *spec.RoomID, userID *spec.UserID) (bool, error) {
+	pending, _, _, _, err := helpers.IsInvitePending(ctx, r.DB, roomID.String(), userID.String())
+	return pending, err
+}
+
+func (r *Queryer) QueryRoomInfo(ctx context.Context, roomID *spec.RoomID) (*types.RoomInfo, error) {
+	return r.DB.RoomInfo(ctx, roomID.String())
+}
+
+func (r *Queryer) GetStateEvent(ctx context.Context, roomID *spec.RoomID, eventType spec.MatrixEventType, stateKey string) (gomatrixserverlib.PDU, error) {
+	return r.DB.GetStateEvent(ctx, roomID.String(), string(eventType), "")
+}
+
+func (r *Queryer) IsInRoom(ctx context.Context, roomNID types.RoomNID, userID *spec.UserID) (bool, error) {
+	_, isIn, _, err := r.DB.GetMembership(ctx, roomNID, userID.String())
+	return isIn, err
+}
+
+func (r *Queryer) GetLocallyJoinedUsers(ctx context.Context, roomVersion gomatrixserverlib.RoomVersion, roomNID types.RoomNID) ([]gomatrixserverlib.PDU, error) {
+	joinNIDs, err := r.DB.GetMembershipEventNIDsForRoom(ctx, roomNID, true, true)
+	if err != nil || len(joinNIDs) == 0 {
+		// There should always be more than one join NID at this point
+		// because we are gated behind GetLocalServerInRoom, but y'know,
+		// sometimes strange things happen.
+		return nil, err
```

Comment:
This means it is possible for `GetLocallyJoinedUsers` to return `nil, nil` in the case where `err == nil` and `len(joinNIDs) == 0`. Is this intentional? If so, please comment. I think it probably is intentional as if there are no joined users then we want to return an empty slice with no error?

Reply:
I think the logic should change to remove the `len(joinNIDs) == 0` check altogether, and return an empty slice later on in that case. And this comment appears to be obsolete as well so it should be removed.
That should clean up the logic.